plenum/test/replica/test_update_watermarks_api.py

```diff
@@ -0,0 +1,168 @@
+    replica._primaryName = None
+    replica.isMaster = False
+    replica.catchup_clear_for_backup()
+    assert replica.last_ordered_3pc == (replica.viewNo, 0)
+    assert replica.h == 0
+    assert replica.H == sys.maxsize
+
+def test_reset_watermarks_before_new_view_on_master(replica, tconf):
+    replica.isMaster = True
+    replica.h = 100
+    replica._reset_watermarks_before_new_view()
+    assert replica.h == 0
+    assert replica.H == tconf.LOG_SIZE
+    assert replica._lastPrePrepareSeqNo == replica.h
+
+
+def test_reset_watermarks_before_new_view_non_master(replica, tconf):
+    replica.isMaster = False
+    replica.h = 100
+    replica._reset_watermarks_before_new_view()
+    assert replica.h == 0
+    assert replica.H == tconf.LOG_SIZE
+    assert replica._lastPrePrepareSeqNo == replica.h
+
+
+def test_catchup_without_vc_and_no_primary_on_master(replica, tconf):
+    replica.isMaster = True
+    replica.isMaster = False
+    replica._primaryName = None
+    # next 3 calls emulate a catchup without vc when there is no primary selected
+    # like it will be called for 'update watermark' procedure
+    replica.caught_up_till_3pc((replica.viewNo, 100))
+    assert replica.h == 100
+    assert replica.H == 100 + tconf.LOG_SIZE
+
+
+def test_catchup_without_vc_and_no_primary_on_backup(replica, tconf):
+    replica.isMaster = False
+    replica.isMaster = False
+    replica._primaryName = None
+    # next 3 calls emulate a catchup without vc when there is no primary selected
+    # like it will be called for 'update watermark' procedure
+    replica.catchup_clear_for_backup()
+    replica._reset_watermarks_before_new_view()
+    replica._setup_for_non_master_after_view_change(replica.viewNo)
+    assert replica.h == 0
+    assert replica.H == tconf.LOG_SIZE
+
+
+def test_catchup_without_during_vc_with_primary_on_master(replica, tconf):
+    # this test emulate situation of simple catchup procedure without view_change
+    # (by checkpoints or ledger_statuses)
+    replica._primaryName = 'SomeNode'
+    replica.isMaster = True
+    replica.caught_up_till_3pc((replica.viewNo, 100))
+    assert replica.last_ordered_3pc == (replica.viewNo, 100)
+    assert replica.h == 100
+    assert replica.H == 100 + tconf.LOG_SIZE
+
+
+
+def test_catchup_without_during_vc_with_primary_on_backup(replica):
+    # this test emulate situation of simple catchup procedure without view_change
+    # (by checkpoints or ledger_statuses)
+    replica._primaryName = 'SomeNode'
+    replica.isMaster = False
+    replica.catchup_clear_for_backup()
+    assert replica.last_ordered_3pc == (replica.viewNo, 0)
+    assert replica.h == 0
+    assert replica.H == sys.maxsize
+
+
+def test_view_change_no_propagate_primary_on_master(replica, tconf):
+    replica.isMaster = True
+    replica._primaryName = 'SomeNode'
+    # next 3 calls emulate simple view_change procedure (replica's watermark related steps)
+    replica.caught_up_till_3pc((replica.viewNo, 100))
```

Comment:
There is just one call here. Please create a helper method for emulate view change.
It should call `caught_up_till_3pc` and replica's `primaryName` setter.